staging/src/k8s.io/apiserver/pkg/util/webhook/client.go

```diff
@@ -148,6 +149,11 @@ func (cm *ClientManager) HookClient(cc ClientConfig) (*rest.RESTClient, error) {
 
 		cfg.ContentConfig.NegotiatedSerializer = cm.negotiatedSerializer
 		cfg.ContentConfig.ContentType = runtime.ContentTypeJSON
+
+		// Add a transport wrapper that allows detection of TLS connections to
+		// servers without SAN extension in their serving certificates
+		cfg.Wrap(x509metrics.NewMissingSANRoundTripperWrapperConstructor(x509MissingSANCounter))
```

Comment:
we usually make it pluggable for testing and future extension

Reply:
This code should be removed as soon as the codebase gets compiled with golang version that does not allow working around CN host verification. According to the release notes, that's 1.17. Therefore I wouldn't worry too much about the future extensions of this code.